### ISSUE TITLE
Don't have body parameters in the doc 

### ISSUE THREAD

**Issue by chinhtranvan** [oatpp/oatpp#738]
Hi, 
I have another interesting question,

if the endpoints taking JSON format data (using Body As DTO) instead of multiple query params. e.g.
**curl -X POST -d '{"abc": true}' http://localhost:8000/demo/api/users** 
There's not a variable about params in body JSON => we don't have params in API documentation 

Do we have anyway to have param in body JSON for documentation?

**Comment by lganzzzo** [comment]
Hello @chinhtranvan ,

I'm not sure if I understand you question correctly...

If you define you endpoint like follows:

```cpp
ENDPOINT("POST", "/users", postUsers,
         BODY_DTO(Object<UserDto>, userDto))
{
  ...
  return createResponse(Status::CODE_200, "OK");
}
```

The model of the body object will be documented in the swagger-ui - something like the:

<img width="586" alt="Screenshot 2023-04-22 at 2 56 26 AM" src="https://user-images.githubusercontent.com/5940640/233750360-1e78102e-e41c-49a1-9abc-ccc7897728b3.png">

**Comment by chinhtranvan** [comment]
Hi Leonid,

Thank you for your answer, I mean that I use **objectMapper** to parse the body of a oatpp::web::client::RequestExecutir::Response using method **readBodyToDto**.  
 e.g.,
auto objectMapper = oatpp::parser::json::mapping::ObjectMapper::createShared();
auto dtmResponseDTO = response->readBodyToDto<oatpp::Object<MyDto>>(objectMapper);

My DTO look likes below.
```
class MyDto : public oatpp::DTO {

DTO_INIT(MyDto, DTO)

DTO_FIELD(String, **abc**);
DTO_FIELD(Int32, **def**);

static oatpp:object....{
auto dto = createShared();
return dto
}
};
```
but I didn't see **abc** and **def** in the document. It show empty {}  in the swagger-ui.

**Comment by chinhtranvan** [comment]
Hi Leonid,

Could you take a look my follow up question if you have any chance?

**Comment by lganzzzo** [comment]
Hello @chinhtranvan ,

When using `ENDPOINT_ASYNC` you have to explicitly add `ENDPOINT_INFO`:

```cpp
  ENDPOINT_INFO(PostUser) {
    info->addConsumes<Object<UserDto>>("application/json"); //<--- Consumes UserDto
  }
  ENDPOINT_ASYNC("POST", "with-body", PostUser) {

    ENDPOINT_ASYNC_INIT(PostUser)
    
    Action act() override {
      return _return(...);
    }
    
  }
```

Regards,
Leonid

**Comment by chinhtranvan** [comment]
Yes, Thank you for your advice. 👍 It works with `ENDPOINT_ASYNC`

**Comment by williamwyc** [comment]
A follow up question on this, when using `ENDPOINT_ASYNC`, if I want to add an optional DTO using addConsumes, how do I specify it in the doc as required = false?

**Comment by ezewu** [comment]
ENDPOINT_INFO(PostUser) {
        info->addConsumes<Object<UserDto>>("application/json");
    }
    ENDPOINT_ASYNC("POST", "user", PostUser,BODY_DTO(Object<UserDto>, userDto)) {

        ENDPOINT_ASYNC_INIT(PostUser)

        Action act() override {
           
        }

    }    How to scan and receive json parameters asynchronously
